ffi/capi/src/units_converter.rs

```diff
@@ -0,0 +1,127 @@
+            icu::experimental::units::converter_factory::ConverterFactory::converter,
+            FnInStruct
+        )]
+        pub fn converter(
+            &self,
+            from: &ICU4XMeasureUnit,
+            to: &ICU4XMeasureUnit,
+        ) -> Option<Box<ICU4XUnitsConverter>> {
+            Some(ICU4XUnitsConverter(self.0.converter(&from.0, &to.0)?).into())
+        }
+
+        /// Creates a parser to parse the CLDR unit identifier (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::parser,
+            FnInStruct
+        )]
+        pub fn parser<'a>(&'a self) -> Box<ICU4XMeasureUnitParser<'a>> {
+            ICU4XMeasureUnitParser(self.0.parser()).into()
+        }
+    }
+
+    #[diplomat::opaque]
+    /// An ICU4X Measurement Unit parser object which is capable of parsing the CLDR unit identifier
+    /// (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+    #[diplomat::rust_link(icu::experimental::units::measureunit::MeasureUnitParser, Struct)]
+    #[diplomat::rust_link(
+        icu::experimental::units::measureunit::MeasureUnitParser,
+        Struct,
+        hidden
+    )]
+    pub struct ICU4XMeasureUnitParser<'a>(pub MeasureUnitParser<'a>);
+
+    impl<'a> ICU4XMeasureUnitParser<'a> {
+        /// Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding [`ICU4XMeasureUnit`].
+        /// Returns an error if the unit identifier is not valid.
+        #[diplomat::rust_link(
+            icu::experimental::units::measureunit::MeasureUnitParser::parse,
+            FnInStruct
+        )]
+        pub fn parse_measure_unit(
+            &self,
+            unit_id: &str,
+        ) -> Result<Box<ICU4XMeasureUnit>, ICU4XError> {
+            Ok(Box::new(ICU4XMeasureUnit(
+                self.0.try_from_identifier(unit_id)?,
+            )))
+        }
+    }
+
+    #[diplomat::opaque]
+    /// An ICU4X Measurement Unit object which represents a single unit of measurement
+    /// such as `meter`, `second`, `kilometer-per-hour`, `square-meter`, etc.
+    ///
+    /// You can create an instance of this object using [`ICU4XMeasureUnitParser`] by calling the `parse_measure_unit` method.
+    #[diplomat::rust_link(icu::experimental::units::measureunit::MeasureUnit, Struct)]
+    #[diplomat::rust_link(icu::experimental::units::measureunit::MeasureUnit, Struct, hidden)]
+    pub struct ICU4XMeasureUnit(pub MeasureUnit);
+
+    #[diplomat::opaque]
+    /// An ICU4X Units Converter object, capable of converting between two [`ICU4XMeasureUnit`]s.
+    ///
+    /// You can create an instance of this object using [`ICU4XUnitsConverterFactory`] by calling the `converter` method.
+    #[diplomat::rust_link(icu::experimental::units::converter::UnitsConverter, Struct)]
+    #[diplomat::rust_link(icu::experimental::units::converter::UnitsConverter, Struct, hidden)]
```

Comment:
also a duplicate

You need lines of "hidden" rust_links when the ffi coverage checker is complaining about _other_ paths, and you want to say that the other path is also covered by this type, but you only want one of these paths to show up in the docs.

So you should never have two of these on a type with the same path. There should always be one non-hidden one, and then there _might_ be more _if_ there are other paths to cover. Which happens when we've merged APIs, or have reexports, or if the Rust-side API just has more complexity.

Reply:
Aha ... okay, in this case, we do not need it here